libtransmission/verify.cc

```diff
@@ -30,94 +23,22 @@ auto constexpr SleepPerSecondDuringVerify = 100ms;
 }
 } // namespace
 
-void tr_verify_worker::verify_torrent(Mediator& verify_mediator, std::atomic<bool> const& abort_flag)
+void tr_verify_worker::verify_torrent(Mediator& verify_mediator, bool const abort_flag)
 {
     verify_mediator.on_verify_started();
 
-    tr_sys_file_t fd = TR_BAD_SYS_FILE;
-    uint64_t file_pos = 0U;
-    uint32_t piece_pos = 0U;
-    tr_file_index_t file_index = 0U;
-    tr_file_index_t prev_file_index = ~file_index;
-    tr_piece_index_t piece = 0U;
-    auto buffer = std::vector<std::byte>(1024U * 256U);
-    auto sha = tr_sha1::create();
     auto last_slept_at = current_time_secs();
-
-    auto const& metainfo = verify_mediator.metainfo();
-    while (!abort_flag && piece < metainfo.piece_count())
+    for (tr_piece_index_t piece = 0U, n = verify_mediator.metainfo().piece_count(); !abort_flag && piece < n; ++piece)
     {
-        auto const file_length = metainfo.file_size(file_index);
+        verify_mediator.on_piece_checked(piece, verify_mediator.check_piece(piece));
 
-        /* if we're starting a new file... */
-        if (file_pos == 0 && fd == TR_BAD_SYS_FILE && file_index != prev_file_index)
+        /* sleeping even just a few msec per second goes a long
+         * way towards reducing IO load... */
+        if (auto const now = current_time_secs(); last_slept_at != now)
         {
-            auto const found = verify_mediator.find_file(file_index);
-            fd = !found ? TR_BAD_SYS_FILE : tr_sys_file_open(found->c_str(), TR_SYS_FILE_READ | TR_SYS_FILE_SEQUENTIAL, 0);
-            prev_file_index = file_index;
+            last_slept_at = now;
+            std::this_thread::sleep_for(SleepPerSecondDuringVerify);
         }
-
-        /* figure out how much we can read this pass */
-        uint64_t left_in_piece = metainfo.piece_size(piece) - piece_pos;
-        uint64_t left_in_file = file_length - file_pos;
-        uint64_t bytes_this_pass = std::min(left_in_file, left_in_piece);
-        bytes_this_pass = std::min(bytes_this_pass, uint64_t(std::size(buffer)));
-
-        /* read a bit */
-        if (fd != TR_BAD_SYS_FILE)
-        {
-            auto num_read = uint64_t{};
-            if (tr_sys_file_read_at(fd, std::data(buffer), bytes_this_pass, file_pos, &num_read) && num_read > 0)
-            {
-                bytes_this_pass = num_read;
-                sha->add(std::data(buffer), bytes_this_pass);
-                tr_sys_file_advise(fd, file_pos, bytes_this_pass, TR_SYS_FILE_ADVICE_DONT_NEED);
```

Comment:
This is an important call (see https://github.com/transmission/transmission/commit/122d69d01393b4e8da165ae45e865ae12978e749 (16.12.2009)).

Reply:
Resolving since this is no longer relevant now that this change was reverted.